trio/_socket.py

```diff
@@ -690,77 +848,146 @@ async def connect(self, address):
         # Okay, the connect finished, but it might have failed:
         err = self._sock.getsockopt(_stdlib_socket.SOL_SOCKET, _stdlib_socket.SO_ERROR)
         if err != 0:
-            raise OSError(err, f"Error connecting to {address}: {os.strerror(err)}")
+            raise OSError(err, f"Error connecting to {address!r}: {os.strerror(err)}")
 
     ################################################################
     # recv
     ################################################################
 
+    # Not possible to typecheck with a Callable (due to DefaultArg), nor with a
+    # callback Protocol (https://github.com/python/typing/discussions/1040)
+    # but this seems to work. If not explicitly defined then pyright --verifytypes will
+    # complain about AmbiguousType
     if TYPE_CHECKING:
 
-        async def recv(self, buffersize: int, flags: int = 0) -> bytes:
+        def recv(__self, __buflen: int, __flags: int = 0) -> Awaitable[bytes]:
             ...
 
-    else:
-        recv = _make_simple_sock_method_wrapper("recv", _core.wait_readable)
+    # _make_simple_sock_method_wrapper is typed, so this checks that the above is correct
+    # this requires that we refrain from using `/` to specify pos-only
+    # args, or mypy thinks the signature differs from typeshed.
+    recv = _make_simple_sock_method_wrapper(  # noqa: F811
+        _stdlib_socket.socket.recv, _core.wait_readable
+    )
 
     ################################################################
     # recv_into
     ################################################################
 
-    recv_into = _make_simple_sock_method_wrapper("recv_into", _core.wait_readable)
+    if TYPE_CHECKING:
+
+        def recv_into(
+            __self, buffer: Buffer, nbytes: int = 0, flags: int = 0
```

Comment:
Another pos-only parameter than can use `/`: I won't comment on any others, as you can just ctrl+f "`__`"

Reply:
As I mentioned above (but gosh, navigating discussion threads in long PR's are impossible):
> Changed it in a bunch of places, but I failed to remove them from sendto, and from the ones where _make_simple_sock_method_wrapper is used to verify the signature.

As it checks if the signature is identical to typeshed it will require updating those. I'll add a comment in the code

Reply:
Oh, I do have a comment for that already.
https://github.com/python-trio/trio/blob/8a8b0b7964e870a0d22963805677065827b65a39/trio/_socket.py#L866-L868